Sources/StreamChat/StateLayer/ChatClient+ChatList.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - Factory Methods for Creating Chat Lists
+
+@available(iOS 13.0, *)
+extension ChatClient {
+    /// Creates an instance of ``ChatList`` which represents an array of channels matching to the specified ``ChannelListQuery``.
```

Comment:
We are talking about channels, but it creates a "ChatList". It is probably better to rename this to `ChannelList` and maybe even rename `Chat` to `Channel` 🤔 It will probably be confusing to old customers who migrate and we have all these new names 🤔 

We need to clarify the namings since it can be confusing:
- Channel List or Chat List
- Channel or Chat or MessageList
- Thread or MessageList

One idea for UI:
- MessageList: The tables of messages.
- Chat: A message list and a composer tied together.
- Channel: A Chat in a channel.
- Thread: A Chat in a thread.

So in terms of LLC, a Chat, could be a Low level component that is shared between the Channel and the Thread.

Reply:
yes, I also think `ChannelList` makes more sense. Not sure I agree about `Chat`, as an interactor object that is used in the whole chat view, might make sense to keep it (let's discuss it).